_posts/2021-12-3-arrow-duckdb.md

```diff
@@ -0,0 +1,383 @@
+  summarise(
+    fare_amount = mean(fare_amount, na.rm = TRUE),
+    tip_amount = mean(tip_amount, na.rm = TRUE),
+    tip_pct = mean(tip_pct, na.rm = TRUE)
+  ) %>%
+  arrange(passenger_count) %>%
+  collect()
+```
+
+### Python
+The workflow in Python is as simple as it is in R. In this example we use DuckDB's Relational API.
+
+``` python
+import duckdb
+import pyarrow as pa
+import pyarrow.dataset as ds
+
+# Open dataset using year,month folder partition
+nyc = ds.dataset('nyc-taxi/', partitioning=["year", "month"])
+
+# We transform the nyc dataset into a DuckDB relation
+nyc = duckdb.arrow(nyc)
+
+# Run same query again
+nyc.filter("year > 2014 & passenger_count > 0 & trip_distance > 0.25 & fare_amount > 0")
+    .aggregate("SELECT AVG(fare_amount), AVG(tip_amount), AVG(tip_amount / fare_amount) as tip_pct","passenger_count").arrow()
+```
+
+## DuckDB and Arrow: The Basics
+
+In this section, we will look at some basic examples of the code needed to read and output Arrow tables in both Python and R.
+
+#### Setup
+
+First we need to install DuckDB and Arrow. The installation process for both libraries in Python and R is shown below.
+```bash
+# Python Install
+pip install duckdb
+pip install pyarrow
+```
+
+``` R
+# R Install
+install.packages("duckdb")
+install.packages("arrow")
+```
+
+To execute the sample-examples in this section, we need to download the following custom parquet files:
+ - https://github.com/duckdb/duckdb-web/blob/master/_posts/data/integers.parquet?raw=true
+ - https://github.com/cwida/duckdb-data/releases/download/v1.0/lineitemsf1.snappy.parquet
+
+
+#### Python
+
+There are two ways in Python of querying data from Arrow:
+1. Through the Relational API
+```py
+# Reads Parquet File to an Arrow Table
+arrow_table = pq.read_table('integers.parquet')
+
+# Transforms Arrow Table -> DuckDB Relation
+rel_from_arrow = duckdb.arrow(arrow_table)
+
+# we can run a SQL query on this and print the result
+print(rel_from_arrow.query('arrow_table', 'SELECT SUM(data) FROM arrow_table WHERE data > 50').fetchone())
+
+# Transforms DuckDB Relation -> Arrow Table
+arrow_table_from_duckdb = rel_from_arrow.arrow()
+```
+
+2. By using replacement scans and querying the object directly with SQL:
+```py
+# Reads Parquet File to an Arrow Table
+arrow_table = pq.read_table('integers.parquet')
+
+# Gets Database Connection
+con = duckdb.connect()
+
+# we can run a SQL query on this and print the result
+print(con.execute('SELECT SUM(data) FROM arrow_table WHERE data > 50').fetchone())
+
+# Transforms Query Result from DuckDB to Arrow Table
+# We can directly read the arrow object through DuckDB's replacement scans.
+con.execute("SELECT * FROM arrow_table").fetch_arrow_table()
+```
+
+It is possible to transform both DuckDB Relations and Query Results back to Arrow.
+
+#### R
+
+In R, the primary way to interact with Arrow is by registering the table as a view (an alternative is to use dplyr as shown above).
```

Comment:
Is this "the primary way"?

Reply:
I read this as in opposition to the Python Relational API, though agree we should tweak the wording to make that clear